### PR TITLE
Add collectibles model

### PR DESCRIPTION
**Pull request by vikmeup**
Closes: https://github.com/TrustWallet/blockatlas/issues/25

### CODE REVIEW

File: collectibles.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package blockatlas
+
+type Collection struct {
+	Name string `json:"name"`
+	Symbol string `json:"symbol"`
+	ImageUrl string `json:"image_url"`
+	Description string `json:"description"`
+	External_Link string `json:"external_link"`
```
Comment (terorie) [inline collectibles.go]: Underscores are quite uncommon in exported Go fields, I think it should be `ExternalLink`

File: collectibles.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+type Collectible struct {
+	Token_ID string `json:"token_id"`
+	Contract_Address string `json:"contract_address"`
+	Category string `json:"category"`
+	Image_URL string `json:"image_url"`
+	External_Link string `json:"external_link"`
+	Type string `json:"type"`
+	Description string `json:"type"`
```
Comment (vikmeup) [inline collectibles.go]: should be `json:"description"` here

File: collectibles.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package blockatlas
+
+type Collection struct {
+	Name            string `json:"name"`
+	Symbol          string `json:"symbol"`
+	ImageUrl        string `json:"image_url"`
```
Comment (terorie) [inline collectibles.go]: Inconsistent naming: `ImageUrl` here

File: collectibles.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	Version         string `json:"version"`
+	Coin            int    `json:"coin"`
+	Type            string `json:"type"`
+}
+
+type Collectible struct {
+	TokenID         string `json:"token_id"`
+	ContractAddress string `json:"contract_address"`
+	Category        string `json:"category"`
+	ImageURL        string `json:"image_url"`
```
Comment (terorie) [inline collectibles.go]: ... But `ImageURL` here

File: cmd/api/generic.go [hunk 5]
```diff
@@ -84,6 +84,35 @@ func makeStakingRoute(router gin.IRouter, api blockatlas.Platform) {
...
+		collections, err := collectionAPI.GetCollections(c.Param("owner"))
+
+		if err != nil {
+			c.JSON(http.StatusServiceUnavailable, err)
+		}
+
+		c.JSON(http.StatusOK, collections)
+	})
+
+	router.GET("/collections/:owner/contract/:contract", func(c *gin.Context) {
```
Comment (vikmeup) [inline cmd/api/generic.go]: `contract` to `collectible` ?

File: cmd/api/generic.go [hunk 6]
```diff
@@ -84,6 +84,35 @@ func makeStakingRoute(router gin.IRouter, api blockatlas.Platform) {
...
+		collectibles, err := collectionAPI.GetCollectibles(c.Param("owner"), c.Param("contract"))
+
+		if err != nil {
+			c.JSON(http.StatusServiceUnavailable, err)
```
Comment (vikmeup) [inline cmd/api/generic.go]: should we add generic function to return error? @terorie  ?

**Comment by madcake** [comment]
@terorie if you choose change request you should refresh review or you will block other contributors.

**Pull request merged**